elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java

```diff
@@ -327,11 +329,30 @@ protected RequestScope(String path, JsonApiDocument jsonApiDocument, RequestScop
      * @return The global filter expression evaluated at the first load
      */
     public Optional<FilterExpression> getLoadFilterExpression(Class<?> loadClass) {
-        if (globalFilterExpression == null) {
-            String typeName = dictionary.getJsonAliasFor(loadClass);
-            return getFilterExpressionByType(typeName);
+        Optional<FilterExpression> permissionFilter;
+        try {
+            permissionFilter = getPermissionExecutor().getReadPermissionFilter(loadClass);
+        } catch (ForbiddenAccessException e) {
+            permissionFilter = Optional.empty();
+        }
+        if (!permissionFilter.isPresent()) {
+            if (globalFilterExpression == null) {
```

Comment:
I would pull getting the global filter expression out of this if statement - so we don't duplicate this logic.